src/3rd_party_bundle_add.c

```diff
@@ -120,14 +123,71 @@ static enum swupd_code validate_binary(struct file *file)
 	return ret_code;
 }
 
+static enum swupd_code validate_permissions(struct file *file)
+{
+	enum swupd_code ret_code = SWUPD_OK;
+	struct stat file_stats;
+	char *staged_file = NULL;
+
+	if (!file) {
+		return ret_code;
```

Comment:
I'm not sure on the usage but if the file doesn't exist this return SWUPD_OK, is that intended?

Reply:
yeah, that was intended.
This is a condition that really will never happen since this function is being called from a loop that is going through existing files... but I decided to go with SWUPD_OK because this function is supposed to perform some validation on a "file", but if "file" is NULL is really because we passed NULL to the function, not because the file doesn't meet the validation we are doing, so it felt more appropriate to just return SWUPD_OK.